### ISSUE TITLE
Enriched rendering of URLs

### ISSUE THREAD

**Issue by Crissov** [whatwg/url#417]
The [rendering section](https://url.spec.whatwg.org/#url-rendering) leaves unmentioned and hence unspecified where this rendering might and will occur. A frequent place is the address bar in web browsers or file managers. The prose currently suggests:

- Show the serialized URL.
- Don't show credentials for [spoofing protection](https://url.spec.whatwg.org/#security-considerations). (Obviously does not apply to email addresses etc.)
- Show characters instead of their codes for usability.
- Use LTR embedding.
- Only show host if necessary (which it is in web browsers).

Some applications transform or style the plain URL in more ways or deviate from the recommendations paraphrased above:

- Show codes instead of characters if they don't come all from the same script, favoring spoof protection over readability.
- Hide the scheme (e.g. `file:`) or emphasize whether it's using encryption (`http` vs. `https`).
- Use friendly, localized short names (e.g. Apple Finder, Windows Explorer; email addresses).
- Make path segments interactive navigation elements (e.g. Windows Explorer).
- Remove frequent or redundant parts (e.g. `www.` in Safari or `m.`  in Chrome 69).

Other _rich_ renderings can be imagined, e.g.:

- Replace ccTLDs by (emoji) flags.
- Deemphasize known public suffixes from [PSL](https://publicsuffix.org/).
- Replace company names by logos (e.g. from favicons).
- Make search parameters interactive if [Open Search](https://github.com/dewitt/opensearch/blob/master/opensearch-1-1-draft-6.md#opensearch-url-template-syntax) metadata is available. Also #192 (RFC6570 templates)
- Make fragment identifier interactive based upon document content. Also #389 (fragment parameters)

In conclusion, I believe this section needs some expansion.

**Comment by annevk** [comment]
FWIW, 8809598ddfd1d935432c8a0cad53f13d70e24bc6 changed this section quite a bit.

**Comment by annevk** [comment]
I think I'm happy with this section now. What do you think @Crissov?
